src/slack.coffee

```diff
@@ -242,4 +245,29 @@ class SlackBot extends Adapter
     channel = @client.getChannelGroupOrDMByName envelope.room
     channel.setTopic strings.join "\n"
 
+  customMessage: (data) =>
+    channel = @client.getChannelGroupOrDMByName data.message.envelope.room
+
+    msg = []
+    data.attachments = data.content if data.content and not data.attachments
+    data.attachments = [data.attachments] unless Array.isArray data.attachments
+
+    msg.attachments = []
+    for item in data.attachments
+      msg.attachments.push item
+
+    msg.text = data.text
+
+    if data.username && data.username != robot.name
+      msg.as_user = false
+      if data.icon_url?
+        msg.icon_url = data.icon_url
+      else if data.icon_emoji?
+        msg.icon_emoji = data.icon_emoji
```

Comment:
Unless it's getting the user name from the authentication token, which might make sense, but explicitly setting the username makes even more sense, IMO.